app/controllers/tracks/exercises_controller.rb

```diff
@@ -45,7 +45,7 @@ def start
   def edit
     return if @solution
 
-    if @user_track && !@user_track.external?
+    if !@user_track.external?
```

Comment:
```suggestion
    unless @user_track.external?
```

Reply:
In fact, change this to be a guard `redirect_to action: :show if @user_track.external?`